peachjam/models/core_document_model.py

```diff
@@ -155,6 +155,18 @@ def clean(self):
         except ValueError:
             raise ValidationError("Invalid FRBR URI: " + frbr_uri)
 
+        self.assign_frbr_uri()
```

Comment:
Assigning it sets it on self so there's no reason to call generate

Also double check what will happen if the date field is missing or similar, which prevents the frbr uri from being assigned

Reply:
@longhotsummer If you save without a frbr_uri date, it pulls the document date and uses that to create the frbr_uri. It allows the frbr_uri actor and subtype to be empty. 

> Assigning it sets it on self so there's no reason to call generate

I have it this way because the generate_expression_frbr function uses self.work_frbr_uri internally. We could pass it in as a parameter and have it work independently, if that would be a bit cleaner.
Also we are doing the same thing inside of save because the model.clean method gets called when saving using model forms. I may be mistaken. Could you confirm if this is the case?

Reply:
You're right, I misread. `assign_frbr_uri()` doesn't touch the expression FRBR URI.